### ISSUE TITLE
EPCOT-COP model training

### ISSUE THREAD

**Issue by AayushGrover** [liu-bioinfo-lab/EPCOT#6]
Hello,

Thank you for the amazing work.

I want to retrain the COP model (5Kb resolution) using only GM12878 and K562 cell lines and test this model on the IMR90 and HepG2 cell lines.

1. To ensure that there is no data leakage, I first trained the pretraining network after changing the list of cell lines in [pretraining/backbone_train.py](https://github.com/liu-bioinfo-lab/EPCOT/blob/main/pretraining/backbone_train.py), [pretraining/pre_train.py](https://github.com/liu-bioinfo-lab/EPCOT/blob/main/pretraining/pre_train.py), and [pretraining/dataset.py](https://github.com/liu-bioinfo-lab/EPCOT/blob/main/pretraining/dataset.py) as well as fixing data paths. The datasets are downloaded from the [Google Drive](https://drive.google.com/drive/folders/1gsveyTgYwlXK5Ntnx5nLKSzIW3JvxLse?usp=share_link) or [Zenodo](https://doi.org/10.5281/zenodo.7485616). 
   ```bash
   python pretraining/backbone_train.py
   python pretraining/pre_train.py --batchsize 8
   ```
   The final pretraining model converged at a validation loss of ~0.025.

2. I am then training the two EPCOT-COP models (one for GM12878 and the other for K562) using the pretrained network above and default hyperparameters. The Hi-C files are prepared as described [here](https://github.com/liu-bioinfo-lab/EPCOT/tree/main/COP#data-preparing). Specifically, I run the command below.
   ```bash
   python COP/downstream_train_hic.py --batchsize 1 --pretrain_path "models/pretrain_dnase_norm_GM12878_K562.pt" --fine_tune
   ```
   The best model, after 40 epochs, results in a test Pearson's correlation of ~0.2-0.3, which is significantly lower than the expected value of ~0.7-0.8 mentioned in the manuscript (Figure S11). 

Could you please help me understand the issue with my training process? It would also be extremely helpful if you could add a README on how to train the pretraining and downstream models.

**Comment by linghuajiang** [comment]
Hi, thank you for your interest in EPCOT. Would you mind testing the performance of backbone trained on epigenomic data and sharing the pearson correlation with me? I’d like to check whether the pretraining achieved good performance before you started fine-tuning on COP. 

I also wanted to mention that we have released an updated version, EPCOT v2, which was trained on a larger and more diverse dataset and supports additional modalities. The EPCOT v2 paper is available here:https://academic.oup.com/nar/article/53/21/gkaf1269/8340987, and the GitHub repository is:https://github.com/liu-bioinfo-lab/general_AI_model. It also includes COP task, in case you’d like to take a look.

**Comment by AayushGrover** [comment]
Hi,

Thank you for your response. I tested both the backbone and the pre-training models. For both models, I am reporting validation Pearson's and Spearman's correlation averaged across all 245 epigenomic tracks. 

Backbone model:
Validation loss = 0.015
Validation Pearson's correlation = 0.158 +- 0.098
Validation Spearman's correlation = 0.109 +- 0.056

Pre-training model (without loading backbone):
Validation loss = 0.025
Validation Pearson's correlation = 0.171 +- 0.088
Validation Spearman's correlation = 0.099 +- 0.053

Pre-training model (with loading backbone):
Validation loss = 0.032
Validation Pearson's correlation = 0.219 +- 0.100
Validation Spearman's correlation = 0.084 +- 0.041

I assume there is some problem with the backbone training itself, then. Here, I explain the steps I followed in more detail:

1. I obtained the GM12878 and K562 ATAC-seq pickle files from [Google Drive](https://drive.google.com/drive/folders/1REE056hPXSTOt2nK6B6bpkwnnofAUdsE). 
2. I then generated the train_seq and train_dnase files as described in [this issue](https://github.com/liu-bioinfo-lab/EPCOT/issues/3#issuecomment-1777801529), whereas I obtained the epigenome labels from [Zenodo](https://zenodo.org/records/7485616). 
3. I then changed the list of cell lines in `pretraining/backbone_train.py`, `pretraining/pre_train.py`, and `pretraining/dataset.py`.
4. Finally, I ran the following scripts to pretrain
   ```bash 
    python pretraining/backbone_train.py
    python pretraining/pre_train.py --load_backbone --batchsize 8
    ```

Thank you for your time, and looking forward to your guidance. 

PS: I looked at EPCOT v2, which is even more impressive. I have opened an issue there regarding the availability of processed input and target datasets.

**Comment by linghuajiang** [comment]
It looks like there may be a mismatch between the label file and sequence/ATAC data during your backbone pretraining step. Could you please try downloading the training data from the Google Drive link below and use it to retrain the backbone?
https://drive.google.com/drive/folders/1via8u06Cte1nbFUgeQU_dHTad8vDB-Vw?usp=drive_link

The train_seqs.pickle file only contains sequences for K562, but since the DNA sequence comes from the reference genome, it is identical across all training cell lines. You can reuse it for other cell lines.
